crates/bevy_remote/src/builtin_methods.rs

```diff
@@ -786,25 +777,58 @@ pub fn process_remote_query_request(In(params): In<Option<Value>>, world: &mut W
     let mut response = BrpQueryResponse::default();
     let mut query = query.build();
     for row in query.iter(world) {
-        // The map of component values:
-        let components_map = build_components_map(
-            row.clone(),
-            paths_and_reflect_components.iter().copied(),
-            &type_registry,
-        )
-        .map_err(BrpError::component_error)?;
+        let entity_id = row.id();
+        let entity_ref = world.get_entity(entity_id).expect("Entity should exist");
+
+        let mut components_map = HashMap::new();
+        if components.is_empty() {
```

Comment:
Could we put the stuff inside this `if` scope into an own function for readability?